pallets/schemas/src/lib.rs

```diff
@@ -306,20 +324,54 @@ pub mod pallet {
 		/// * [`Error::InvalidSchema`] - Schema is malformed in some way
 		/// * [`Error::SchemaCountOverflow`] - The schema count has exceeded its bounds
 		#[pallet::call_index(3)]
-		#[pallet::weight(T::WeightInfo::create_schema_via_governance(model.len() as u32))]
+		#[pallet::weight(T::WeightInfo::create_schema_via_governance(model.len() as u32+ settings.len() as u32))]
 		pub fn create_schema_via_governance(
 			origin: OriginFor<T>,
 			creator_key: T::AccountId,
 			model: BoundedVec<u8, T::SchemaModelMaxBytesBoundedVecLimit>,
 			model_type: ModelType,
 			payload_location: PayloadLocation,
+			settings: BoundedVec<SchemaSetting, T::MaxSchemaSettingsPerSchema>,
 		) -> DispatchResult {
 			T::CreateSchemaViaGovernanceOrigin::ensure_origin(origin)?;
-			let schema_id = Self::create_schema_for(model, model_type, payload_location)?;
+			let schema_id = Self::create_schema_for(model, model_type, payload_location, settings)?;
 
 			Self::deposit_event(Event::SchemaCreated { key: creator_key, schema_id });
 			Ok(())
 		}
+
+		/// Adds a given schema to storage with additional settings available from `SchemaSetting`
+		/// # Arguments
+		/// * `origin` - The origin of the call
+		/// * `model` - The schema model
+		/// * `model_type` - The schema model type
+		/// * `payload_location` - The schema payload location
+		/// * `settings` - The bounded list of additional schema settings.
+		///
+		/// # Events
+		/// * [`Event::SchemaCreated`]
+		///
+		/// # Errors
+		/// * [`Error::LessThanMinSchemaModelBytes`] - The schema's length is less than the minimum schema length
+		/// * [`Error::ExceedsMaxSchemaModelBytes`] - The schema's length is greater than the maximum schema length
+		/// * [`Error::InvalidSchema`] - Schema is malformed in some way
+		/// * [`Error::SchemaCountOverflow`] - The schema count has exceeded its bounds
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::WeightInfo::create_schema(model.len() as u32 + settings.len() as u32))]
+		pub fn create_schema_with_settings(
+			origin: OriginFor<T>,
+			model: BoundedVec<u8, T::SchemaModelMaxBytesBoundedVecLimit>,
+			model_type: ModelType,
+			payload_location: PayloadLocation,
+			settings: BoundedVec<SchemaSetting, T::MaxSchemaSettingsPerSchema>,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
```

Comment:
Why is there a separate extrinsic for this, that allows anyone with enough token to create a schema?  It's the same as the governance-based one except for the origin requirement.  Was it left over from before all the governance-based schema creation was done?

Reply:
We added this to not break downstream consumers like saas team.

To have a separate extrinsic to create schema with settings

And thank you bringing this, governance based schema creation was done while this work was ongoing.

Do we need to do anything else with this extrinsic? as this will also be called via governance.

Reply:
As Puneet mentioned we are not that involved with the governance related changes but the main reason to add this was to not break backwards compatibility of existing extrinsic. Would like to know how this would work with governance

Reply:
Looking at code again, looks like create_schema_by_goverance and this new one, does the same thing, so we can get rid of this one

Will push a commit